core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java

```diff
@@ -129,7 +133,9 @@ public void close() {
       HEARTBEAT_CANCEL_POOL.submit(new Runnable() {
         @Override
         public void run() {
+          NUM_PENDING_HEARTBEAT_CLOSE.incrementAndGet();
           mHeartbeat.cancel(true);
+          NUM_PENDING_HEARTBEAT_CLOSE.decrementAndGet();
```

Comment:
Instead of keeping track ourselves, is it possible to just call the appropriate graceful shutdown method of the heartbeat pool?